test/torchtext_unittest/models/models_gpu_test.py

```diff
@@ -1,11 +1,11 @@
+import pytest
 import torch
 
-from ..common.case_utils import skipIfNoCuda
 from ..common.torchtext_test_case import TorchtextTestCase
 from .models_test_impl import BaseTestModels
 
 
-@skipIfNoCuda
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="CUDA is not available")
```

Comment:
Any reason we're no longer using a custom `skipIfNoCuda` decorator to do this check? Wouldn't it be cleaner for the future when we have many gpu specific unit tests?

Reply:
Is the issue with GPU test not executed resolved?
If not then this appears to be no1 suspect for me. 

The unit test suite in torchaudio (which torchtext’s suite is based off of), was originally (and still) written with Python unittest module. No pytest code is used. (pytest command is used to run the code though) The reason for that is 1. to reuse the helper functions from PyTorch and 2. fbcode was not supporting pytest. 

One of the biggest downsides is that ‘unittest’ does not natively support parameterization. So we used ‘parameterized’. ‘parameterized’ (and ‘pytest’ as well) employs many tricks to manipulate meta data and class/type objects, which we wouldn’t (and shouldn’t ) do in normal circumstances writing a library, so they often show strange behavior. 

So far unittest + parameterized for writing tests then using pytest for running them worked fine. 
But this one seems to be combining three to write tests, which we avoided in audio repo and could have some unexpected effects under the hood.

Reply:
Yea using skipIfNoCuda was not working properly, but using pytest.mark.skipif (as vision uses) works fine.

Reply:
torchvision's test suites are based on pytest. So that's a bit different.